cluster/cluster_fail.go

```diff
@@ -655,6 +653,90 @@ func (cluster *Cluster) MasterFailover(fail bool) bool {
 	return true
 }
 
+// FailoverExtraMultiSource care of master extra muti source replications
+func (cluster *Cluster) FailoverExtraMultiSource(oldMaster *ServerMonitor, NewMaster *ServerMonitor, fail bool) error {
+
+	for _, rep := range oldMaster.Replications {
+
+		if rep.ConnectionName.String != cluster.Conf.MasterConn {
+			myparentrplpassword := ""
+			parentCluster := cluster.GetParentClusterFromReplicationSource(rep)
+			cluster.LogPrintf(LvlInfo, "Failover replication source %s ", rep.ConnectionName.String)
+			if parentCluster != nil {
+				myparentrplpassword = parentCluster.rplPass
+			} else {
+				cluster.LogPrintf(LvlErr, "Unable to found a monitored cluster for replication source %s ", rep.ConnectionName.String)
+				cluster.LogPrintf(LvlErr, "Moving source %s with empty password to preserve replication stream on new master", rep.ConnectionName.String)
+			}
+			// need a way to found parent replication password
+			if strings.ToUpper(rep.UsingGtid.String) == "NO" {
+
+				logs, err := dbhelper.ChangeMaster(NewMaster.Conn, dbhelper.ChangeMasterOpt{
```

Comment:
I would recommend creating a `dbhelper.ChangeMasterOpt{}` above the conditionals that share all the properties and then simply refer to that one and add the required parameters that are missing and/or different.

There's a LOT of duplication.